libraries/SRV_Channel/SRV_Channels.cpp

```diff
@@ -209,7 +214,13 @@ void SRV_Channels::setup_failsafe_trim_all_non_motors(void)
 void SRV_Channels::calc_pwm(void)
 {
     for (uint8_t i=0; i<NUM_SERVO_CHANNELS; i++) {
-        channels[i].calc_pwm(functions[channels[i].function].output_scaled);
```

Comment:
take a semphore at top of calc_pwm()

Reply:
Does a semaphore already exist for this? Would it go in the constructor?

Reply:
no need for a constructor for it, just declare like this:
  HAL_Semaphore calc_sem;
in the class header.
To use, do this:
  WITH_SEMAPHORE(calc_sem);